src/metabase/models/humanization.clj

```diff
@@ -82,10 +83,15 @@
   :visibility :settings-manager
   :audit      :raw-value
   :getter     (fn []
-                (let [strategy (setting/get-value-of-type :keyword :humanization-strategy)]
-                  ;; actual advanced method has been excised. Use `:simple` instead if someone had specified
-                  ;; `:advanced`.
-                  (if (= strategy :advanced)
-                    :simple
-                    strategy)))
+                (let [strategy (setting/get-value-of-type :keyword :humanization-strategy)
+                      valid-values (set (keys (methods u.humanization/name->human-readable-name)))
+                      valid-strategy? (contains? valid-values strategy)
+                      ]
+                  (cond
+                    (and (= strategy :advanced) valid-strategy?) :simple
+                    valid-strategy? strategy
+
+                    :else
+                    (do (log/warn (u/format-color :yellow "Invalid humanization strategy '%s'" strategy))
+                        :simple ))))
```

Comment:
I think it might be a little clearer to do the warning up front. Just reduces the number of logic branches a bit.
```suggestion
                  (when (not valid-strategy?) (log/warn ...)
                  (if valid-strategy? strategy :simple)
```